src/CqlBuilderPanel/Includes/Results/IncludeResultActions.tsx

```diff
@@ -18,37 +23,66 @@ const IncludeResultActions = ({
   onEdit,
   onDelete,
   onView,
+  operation,
 }: PropTypes) => {
   return (
     <Stack direction="row" alignItems="center">
       {canEdit && (
         <>
           {showDeleteAction && (
-            <Button
-              variant="outline"
-              data-testid={`delete-button-${id}`}
-              disabled={false}
-              tw="mr-4"
-              onClick={() => {
-                onDelete(id);
-              }}
-            >
-              Delete
-            </Button>
+            <>
+              <ToolTippedIcon
+                tooltipMessage="Delete"
+                buttonProps={{
+                  "data-testid": `delete-button-${id}`,
+                  "aria-label": `delete-button-${id}`,
+                  size: "small",
+                  onClick: () => onDelete(id),
+                }}
+              >
+                <DeleteOutlineIcon color="error" />
+              </ToolTippedIcon>
+              <ToolTippedIcon
+                tooltipMessage="Edit"
+                buttonProps={{
+                  "data-testid": `edit-button-${id}`,
+                  "aria-label": `edit-button-${id}`,
+                  size: "small",
+                  onClick: () => onEdit(id, true),
+                }}
+              >
+                <BorderColorOutlinedIcon color="primary" />
+              </ToolTippedIcon>
+            </>
           )}
-          <Button
-            variant="outline"
-            data-testid={`edit-button-${id}`}
-            disabled={false}
-            tw="mr-4"
-            onClick={() => {
-              onEdit(id, canEdit);
-            }}
-          >
-            View / Apply
-          </Button>
         </>
       )}
+      {operation === "edit" && (
+        <ToolTippedIcon
+          tooltipMessage="View"
+          buttonProps={{
+            "data-testid": `view-button-${id}`,
+            "aria-label": `view-button-${id}`,
+            size: "small",
+            onClick: () => onView(id),
+          }}
+        >
+          <CodeOffOutlinedIcon color="primary" />
+        </ToolTippedIcon>
+      )}
+      {!showDeleteAction && (
+        <Button
+          variant="outline"
+          data-testid={`edit-button-${id}`}
```

Comment:
data-testid={`edit-button-${id}`} --> says edit-button, but the operation that we are doing here is delete. I think this should be changed to data-testid={`delete-button-${id}`}